internal/builder/builder.go

```diff
@@ -103,17 +103,26 @@ type moduleWithDiffID struct {
 type BuilderOption func(*options) error
 
 type options struct {
-	flatten bool
-	exclude []string
+	flatten  bool
+	exclude  []string
+	runImage string
+}
+
+func WithRunImage(name string) BuilderOption {
+	return func(o *options) error {
+		o.runImage = name
+		return nil
+	}
 }
 
 // FromImage constructs a builder from a builder image
-func FromImage(img imgutil.Image) (*Builder, error) {
-	return constructBuilder(img, "", true)
+func FromImage(img imgutil.Image, runImage string) (*Builder, error) {
+	return constructBuilder(img, "", true, WithRunImage(runImage))
```

Comment:
I think we don't need to add this argument here, that's why you need to change a lot of places in the tests with an empty string